packages/components/src/notice/list.tsx

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import Notice from '.';
+import type { WordPressComponentProps } from '../ui/context';
+import type { NoticeListProps } from './types';
+
+const noop = () => {};
+
+function NoticeList( {
+	notices,
+	onRemove = noop,
+	className,
+	children,
+}: WordPressComponentProps< NoticeListProps, 'div', false > ) {
+	const removeNotice =
+		( id: NoticeListProps[ 'notices' ][ number ][ 'id' ] ) => () =>
+			onRemove( id );
+
+	className = classnames( 'components-notice-list', className );
+
+	return (
+		<div className={ className }>
+			{ children }
+			{ [ ...notices ].reverse().map( ( notice ) => {
+				const { content, ...restNotice } = notice;
+				return (
+					<Notice
+						{ ...restNotice }
+						key={ notice.id }
+						onRemove={ removeNotice( notice.id ) }
+					>
+						{ notice.content }
+					</Notice>
+				);
+			} ) }
+		</div>
+	);
+}
+
+export default NoticeList;
```

Comment:
I went with a subcomponent approach, let me know what you think.

Also, I thought it would be helpful for the notices to actually be dismissible, but I know we aren't currently doing that for any of the other stories. Thoughts?

Reply:
> Also, I thought it would be helpful for the notices to actually be dismissible, but I know we aren't currently doing that for any of the other stories.

I think the difference makes sense. `Notice` itself doesn't have any dismiss behavior built in, but the point of `NoticeList` is to handle the dismissals. So naturally the NoticeList story will include a demonstration of the dismiss behavior.